app/models/archive_endpoint.rb

```diff
@@ -16,6 +16,23 @@ class ArchiveEndpoint < ApplicationRecord
   # TODO: after switching to string, validate that input resolves to class which #is_a class of the right type?
   validates :delivery_class, presence: true
 
+  # for the given druid, which archive endpoints should have archive copies?
+  scope :archive_targets, lambda { |druid|
+    joins(preservation_policies: [:preserved_objects]).where(preserved_objects: { druid: druid })
+  }
+
+  scope :which_have_archive_copy, lambda { |druid, version|
```

Comment:
maybe add comment:
"for a druid, version, which zip_endpoints have  zipped_moab_versions  for the complete_moab?"